### PR TITLE
[WIP] implement manage course (fixes #444)

### LINKED ISSUE #444: Manage course functionality
The "Manage" button in the courses menu should lead to the add courses component, but with the info of the selected course filled in and the save button will update the course info for the selected course rather than add a new course.

### PR DESCRIPTION
**Pull request by ArashDai**
WIP implementing the update course functionality for issue #444

### CODE REVIEW

**Comment by paulbert** [comment]
@ArashDai We're sticking with an older version of npm, so please omit the package-lock file from the PR.

**Comment by ArashDai** [comment]
@nirojdyola All that stuff will be removed from the final version, still WIP,
@paulbert Sounds good

**Comment by ArashDai** [comment]
Know issue IsNameAvailible$() validator not working propperly

**Comment by ArashDai** [comment]
Everything should be working now, ready for review

File: src/app/validators/validator.service.ts [hunk 1]
```diff
@@ -38,5 +38,33 @@ constructor(private couchService: CouchService) {}
       })
     ).debug('Checking uniqueness of ' + fieldName + ' in ' + dbName);
   }
+// isNameAvailible is broken
```
Comment (adepojubosun) [inline src/app/validators/validator.service.ts]: still broken ?

File: src/app/validators/validator.service.ts [hunk 2]
```diff
@@ -39,4 +39,32 @@ constructor(private couchService: CouchService) {}
     ).debug('Checking uniqueness of ' + fieldName + ' in ' + dbName);
   }
 
+  public isNameAvailible$(
```
Comment (dogi) [inline src/app/validators/validator.service.ts]: Is this `$` sign necessary? or what do I miss ...

**Pull request merged**